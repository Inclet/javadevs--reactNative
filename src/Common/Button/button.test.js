import React from "react";
import { shallow } from "enzyme";
import CommonButton from "./index";

describe("Button Component", () => {
  describe("RENDERING", () => {
    let wrapper, props;
    beforeEach(() => {
      props = {
        target: {
          navigate: jest.fn()
        },
        page: "AboutUs",
        label: "Login"
      };
      wrapper = shallow(<CommonButton {...props} />);
    });
    it("Should match Snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("Should navigate to a new page when you click on the button", () => {
      wrapper.find(`[data-navigator="buttonNavigator"]`).simulate("press");
      expect(props.target.navigate).toHaveBeenCalledWith(props.page);
    });

    it("Should add label to the button", () => {
      expect(
        wrapper
          .find(`[data-text="buttonLabel"]`)
          .shallow()
          .text()
      ).toEqual(props.label);
    });
  });
});
