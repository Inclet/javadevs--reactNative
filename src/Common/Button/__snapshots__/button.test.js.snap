// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component RENDERING Should match Snapshot 1`] = `ShallowWrapper {}`;
